src/script_opt/ZAM/OPs/iterations.op

```diff
@@ -87,7 +90,10 @@ internal-op Next-Vector-Blank-Iter-Val-Var
 class Vsb
 eval	NextVectorIterCore($1, $2)
 	if ( Z_IS_MANAGED )
+		{
+		ZVal::DeleteManagedType($$);
 		$$ = BuildVal(vv[$1.iter]->ToVal(Z_TYPE), Z_TYPE);
+		}
```

Comment:
Similarly to #4016, it's probably worth adding a test in `opt` so it's covered? Or some test in core or language that would exercise this, if we don't have it?

Running `btest -a zam core language` with an ASAN build on Linux (including leak sanitizer) with current master shows quite a few errors fwiw (not sure they are fixed by the pending PRs)
```
# core
[  2%] core.div-by-zero [zam] ... failed
[  7%] core.expr-exception [zam] ... failed
[ 11%] core.mmdb.explicit-open [zam] ... failed
[ 14%] core.option-priorities [zam] ... failed
[ 16%] core.option-zeek-done [zam] ... failed
[ 37%] core.when-interpreter-exceptions [zam] ... failed
# language
[  4%] language.any [zam] ... failed
[ 19%] language.conditional-expression [zam] ... failed
[ 20%] language.copy-all-opaques [zam] ... failed
[ 21%] language.copy-all-types [zam] ... failed
[ 50%] language.paraglob [zam] ... failed
[ 51%] language.pattern-tables-when [zam] ... failed
[ 69%] language.returnwhen [zam] ... failed
[ 90%] language.vector [zam] ... failed
[ 94%] language.vector-of-interval-arithmetic [zam] ... failed
[ 98%] language.when-scope [zam] ... failed
```